src/components/editProfessorsModal/hooks/useEditProfessorsModal.ts

```diff
@@ -0,0 +1,184 @@
+
+  const professors: TSelectedProfessor[] = useMemo(() => {
+    if (!listProfessorsResponse || !selectedSubject) return [];
+
+    const notAssignedProfessors = listProfessorsResponse.data.filter(
+      (prof) =>
+        selectedSubject.responsables.find(
+          (resp) =>
+            resp.professor.user.id === prof.user.id &&
+            resp.status.status === ReponsabilityProfessorStatus.APPROVED
+        ) === undefined
+    );
+
+    return notAssignedProfessors.map((professor) => ({
+      id: professor.user.id,
+      name: professor.user.name,
+    }));
+  }, [listProfessorsResponse, selectedSubject]);
+
+  const handleCloseModal = () => {
+    setIsOpen(false);
+    resetStatesAssignProfessor();
+    resetStatesEndResponsability();
+    setSelectedProfessorRemove(undefined);
+    setSelectedProfessorsIds([]);
+  };
+
+  const handleOpenModal = (subject: TCompleteSubject) => {
+    setSelectedSubject(subject);
+    setIsOpen(true);
+  };
+
+  const handleClickRemoveProfessor = (responsible: TSubjectResponsible) => {
+    setSelectedProfessorRemove(responsible);
+    setIsRemoveProfessorModalOpen(true);
+    setIsOpen(false);
+  };
+
+  const handleCloseRemoveProfessorModal = () => {
+    setSelectedProfessorRemove(undefined);
+    setIsRemoveProfessorModalOpen(false);
+    setIsOpen(false);
+  };
+
+  const handleChangeProfessors = (event: SelectChangeEvent<string[]>) => {
+    const {
+      target: { value },
+    } = event;
+
+    setSelectedProfessorsIds(
+      typeof value === 'string' ? value.split(',') : value
+    );
+  };
+
+  const handleAssignProfessorsClick = () => {
+    assignProfessors({
+      professors_ids: selectedProfessorsIds.map((id) => Number(id)),
+      subject_id: selectedSubject?.id as number,
+    });
+  };
+
+  const handleEndResponsabilityClick = (id?: number) => {
+    if (id) {
+      void endResponsability(id);
+    }
+  };
+
+  useEffect(() => {
+    void listProfessors({ quantity: 1000 });
+  }, []);
+
+  useEffect(() => {
+    if (isSuccessAssignProfessor) {
+      showSuccessSnackBar('Professor(es) adicionados');
+      resetStatesAssignProfessor();
+      setSelectedProfessorsIds([]);
+      handleCloseModal();
+      refetchSubject();
+    }
+  }, [isSuccessAssignProfessor]);
+
+  useEffect(() => {
+    if (isSuccessEndResponsability) {
+      showSuccessSnackBar('Professor(a) removido(a)');
+      resetStatesEndResponsability();
+      setSelectedProfessorsIds([]);
+      handleCloseRemoveProfessorModal();
+      refetchSubject();
+    }
+  }, [isSuccessEndResponsability]);
+
+  useEffect(() => {
+    if (errorAssignProfessor) {
+      showErrorSnackBar(`Não foi possível concluir a ação, tente novamente.`);
+      resetStatesAssignProfessor();
+      setSelectedProfessorsIds([]);
+    }
+  }, [errorAssignProfessor]);
+
+  useEffect(() => {
+    if (errorEndResponsability) {
+      showErrorSnackBar(`Não foi possível concluir a ação, tente novamente.`);
```

Comment:
Muda isso aqui pra mostrar a mensagem do erro, pf:

```suggestion
      showErrorSnackBar(`Erro ao remover professor: ${errorEndResponsability.message}`);
```

Tá diferente do Figma mas é pra ser assim. Falei com o chris.

Reply:
Resolvido, mudei o de adicionar professor tbm